### PR TITLE
Add `clear`, `copy` and `popitem` methods to coords, masks, and attrs

### LINKED ISSUE #2863: Yet more dict-methods on Coords, etc
- `clear`
- `popitem`
- `copy`?

Maybe:
- `setdefault`: maybe useful in rare cases but one should not create a large variable first and potentially discard it if the key already exists in the dict

Not:
- `fromkeys`: coord=None makes no sense

### PR DESCRIPTION
**Pull request by nvaytet**
Fix #2863

Also moved and refactored a bunch of tests from `dataset_test.py` that were testing dict methods into `dicts_test.py`.

### CODE REVIEW

File: lib/python/bind_data_array.h [hunk 1]
```diff
@@ -128,6 +128,50 @@ void bind_pop(pybind11::class_<T, Ignored...> &view) {
...
+  });
+}
+
+template <class T, class... Ignored>
+void bind_dict_popitem(pybind11::class_<T, Ignored...> &view) {
+  view.def("popitem", [](T &self) {
+    typename T::key_type key;
+    for (const auto &k : keys_view(self))
+      key = k;
+    return py::cast(self.extract(key));
```
Comment (jl-wynen) [inline lib/python/bind_data_array.h:148]: `popitem` should return a tuple of `(key, value)`

**Review by jl-wynen — CHANGES_REQUESTED** [review]

**Review by jl-wynen — COMMENTED** [review]
Please run the stubgen because you made interface changes in C++.

**Comment by nvaytet** [comment]
> Please run the stubgen because you made interface changes in C++.

Postponing until stub generator is fixed.

**Comment by nvaytet** [comment]
@jl-wynen I finally got round to re-running the stubgen. Can you take a final look? thanks

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**